masterbase/guards.py

```diff
@@ -85,3 +85,9 @@ async def valid_session_guard(connection: ASGIConnection, _: BaseRouteHandler) -
         Server.query_from_params(api_key, converted_fake_ip, fake_port)
     except KeyError:
         raise NotAuthorizedException("Cannot accept data from a non-existent gameserver!")
+
+
+async def valid_target_guard(connection: ASGIConnection, _: BaseRouteHandler):
+    """Verify the existence of a target_steam_id."""
+    params = connection.query_params
+    return account_exists(params["target_steam_id"])
```

Comment:
i think we want to change this to something like if not account_exists(...): raise some error....unsure on what ill sleep on it https://umbraco.com/knowledge-base/http-status-codes/#4xx

Reply:
I would say `404`, but the `/report` resource exists, so that wouldn't be why in HTTP parlance. I don't know that status codes close neatly over the cause of failure here. `422` might be a better fit if you want it to be descriptive, `400` if you don't.

Reply:
went with permission denied? `Request understood, but not authorized.` seems correct?